### PR TITLE
feat(docs): interactive documentation knobs

### PR DESCRIPTION
**Pull request by raunofreiberg**
This PR is just a proof of concept for the documentation. It would still require some additional work and refactoring to play nicely.

gif

## Changes
- Use [`react-view`](https://github.com/uber/react-view/) for interactive documentation
- Add `View` component to be used in `.mdx` files in place of `PropsTable`
- Add Button as an example

#### The current `CodeBlock` component does not play nicely with `react-view`
`react-view` has import/export statements whereas `react-live` (used for `CodeBlock`) does not parse these, so the following error is shown:

image

It might be worth dropping the `react-live` one and copying [`Editor` from `react-view`](https://github.com/uber/react-view/blob/master/src/ui/editor.tsx), which uses the same dependencies under the hood.

Another possibility is to transform the code of `react-view` into a more suitable form for `react-live`.

#### Use `PropTypes` from `react-view` in `.mdx` files
I couldn't get this working on first try via `MDXProvider`, but it should be possible.

#### Handle errors properly per knob
Currently, errors are silently ignored. I think we should come up with inline error styles for both the code editor and knobs.

<hr />

I'm absolutely fine if this PR does not get accepted; but if there's any interest in moving forward with this POC, I am happy to continue 🙃

### CODE REVIEW

File: packages/website/gatsby-node.js [hunk 1]
```diff
@@ -74,3 +74,11 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
     });
   }
 };
+
+exports.onCreateWebpackConfig = ({ actions }) => {
```
Comment (raunofreiberg) [inline packages/website/gatsby-node.js]: I don't know what's up with this. We had to add the same to our config for `react-view` to work. 

re: https://github.com/uber/react-view/pull/18#issuecomment-573157256

**Comment by raunofreiberg** [comment]
CI is failing for some reason that I can't see due to not having access to now.

File: packages/website/src/docs/button.mdx [hunk 2]
```diff
@@ -4,34 +4,43 @@ component: Button
 description: Useful to communicate actions to your users.
 ---
 
-```js live
-<Button>New project</Button>
-```
+import { Button } from '@modulz/radix';
```
Comment (raunofreiberg) [inline packages/website/src/docs/button.mdx]: I don't know why, but the build would fail without an explicit import of the component.

**Comment by raunofreiberg** [comment]
@peduarte Friendly ping 🙃

**Comment by peduarte** [comment]
@raunofreiberg a lot of other reviews to do 😉will get to it!

**Comment by peduarte** [comment]
@raunofreiberg WOW! I looooooove this idea ❤️ 

It has made the overall docs much more interactive, very cool indeed 👌

I think in order for us to get this PR in, we'll need to dedicate quite a bit more effort to make sure it's spot on. Unfortunately though, we're not at a point that we want to invest much time on the docs in this repo. 

As you probably know, Design systems generated via Modulz will be accompanied with a fully documented Style guides, so if we decide to adopt this behaviour it's something that we need to do it somewhere else.

This entire repo will soon be replaced by Modulz, once we port Radix over.

But _thank you_ for sharing your idea 👏 🙂

**Comment by raunofreiberg** [comment]
Totally understandable. Thanks for the feedback!

If by any chance you guys decide to pick this up in the future, I'd be happy to contribute to it 🙃

**Comment by raunofreiberg** [comment]
> As you probably know, Design systems generated via Modulz will be accompanied with a fully documented Style guides, so if we decide to adopt this behaviour it's something that we need to do it somewhere else.

Just thinking aloud here — the generated documentation could possibly also leverage this feature to some extent, enabling all components to have the interactiveness! 🙃

**Comment by peduarte** [comment]
@raunofreiberg 💯

**Pull request closed without merging**